File: armstrong.java
import java.util.*;

public class armstrong 
{
    public static void main(String[] args) 
    {
        int sumcube = 0 ;
        int iTemp=0;
        int idigit=0;
        int ino = 189;

        iTemp=ino;

        while (ino>0)
        {
            idigit=ino%10;

            ino=ino/10;

            sumcube=sumcube+(idigit*idigit*idigit);
            
        }
        
        ino=iTemp;
        if (ino==sumcube)
        {
            System.out.println(ino+"is an armstrong number ");
        }
        else
        {
            System.out.println(ino+"is not an armstrong number");
        }
        
    }
    
}

File: Armstrong.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int n = 186;
    int iTemp = n;

    int idigit=0;
    int iSum=0;


    while (n != 0)
    {
        idigit=n%10;

        iSum=iSum+(idigit*idigit*idigit);

        n=n/10;

    }

    if(iSum==iTemp)
    {
        cout<<"It is a Armstrong number\n";
    }
    else
    {
         cout<<"It is not a Armstrong number\n";   
    }
    
}